nvflare/private/fed/client/client_runner.py

```diff
@@ -406,7 +406,8 @@ def fetch_and_run_one_task(self, fl_ctx) -> (float, bool):
         """
         default_task_fetch_interval = self.default_task_fetch_interval
         self.log_debug(fl_ctx, "fetching task from server ...")
-        task = self.engine.get_task_assignment(fl_ctx)
+        temp_fl_ctx = copy.copy(fl_ctx)
+        task = self.engine.get_task_assignment(temp_fl_ctx)
```

Comment:
why do we need this temp_fl_ctx?

Reply:
The idea is to use this throw away the data added into the fl_ctx during the get_task_assignment() events firing. But thinking more, we should not do that in the platform level, instead the user components should take care of them. I will change it back.